project/Doc.scala

```diff
@@ -82,3 +74,63 @@ object Unidoc {
       file
   }
 }
+
+/**
+ * For projects with few (one) classes there might not be any diagrams.
+ */
+object ScaladocNoVerificationOfDiagrams extends AutoPlugin {
+
+  override def trigger = noTrigger
+  override def requires = Scaladoc
+
+  override lazy val projectSettings = Seq(
+    Scaladoc.validateDiagrams in Compile := false
```

Comment:
Wouldn’t it be simpler to just add this one line to those projects that need it?